Classes/Notifications/NotificationClient.swift

```diff
@@ -59,13 +59,25 @@ final class NotificationClient {
         width: CGFloat,
         completion: @escaping (Result<([NotificationViewModel], Int?)>) -> Void
         ) {
-        githubClient.client.send(V3NotificationRequest(all: all, page: page)) { result in
-            switch result {
-            case .success(let response):
-                let viewModels = CreateViewModels(containerWidth: width, v3notifications: response.data)
-                self.fetchStates(for: viewModels, page: response.next, completion: completion)
-            case .failure(let error):
-                completion(.error(error))
+        if let repo = repo {
+            githubClient.client.send(V3RepositoryNotificationRequest(all: all, owner: repo.owner, repo: repo.name)) { result in
+                switch result {
+                case .success(let response):
+                    let viewModels = CreateViewModels(containerWidth: width, v3notifications: response.data)
+                    self.fetchStates(for: viewModels, page: response.next, completion: completion)
+                case .failure(let error):
+                    completion(.error(error))
+                }
+            }
+        } else {
+            githubClient.client.send(V3NotificationRequest(all: all, page: page)) { result in
```

Comment:
This is not an issue related to this PR, but would't be better to have something like this: `github.client` and not `githubClient.client`?

Reply:
cc @rnystrom

Reply:
Probably, leftover from refactors. I want to revisit the whole `GithubClient` class, maybe change it to like `AppSession` or something. It's really just a grab-bag object for injection.